src/triangle.rs

```diff
@@ -0,0 +1,655 @@
+    }
+
+    /// Expand the triangle by a constant amount (`sizes`) in all directions
+    pub fn inflate(&self, size: impl Into<Size>) -> Self {
+        let size = size.into();
+
+        Self::new(
+            (self.a.to_vec2() + size.to_vec2()).to_point(),
+            (self.b.to_vec2() + size.to_vec2()).to_point(),
+            (self.c.to_vec2() + size.to_vec2()).to_point(),
+        )
+    }
+
+    /// A new [`Triangle`] with each vertecie's ordinates rounded to the nearest integer
+    #[inline]
+    pub fn round(self) -> Self {
+        Self::new(self.a.round(), self.b.round(), self.c.round())
+    }
+
+    /// A new [`Triangle`] with each vertecie's ordinates rounded up to the nearest integer
+    #[inline]
+    pub fn ceil(self) -> Self {
+        Self::new(self.a.ceil(), self.b.ceil(), self.c.ceil())
+    }
+
+    /// A new [`Triangle`] with each vertecie's ordinates rounded down to the nearest integer
+    #[inline]
+    pub fn floor(self) -> Self {
+        Self::new(self.a.floor(), self.b.floor(), self.c.floor())
+    }
+
+    /// A new [`Triangle`],
+    /// with each coordinate value rounded towards the center of the [`Triangle`]
+    /// to the nearest integer, unless they are already an integer.
+    /// That is to say this function will return the biggest possible [`Triangle`]
+    /// with integer coordinates that is a subset of `self`.
+    #[inline]
+    pub fn expand(self) -> Self {
+        Self::new(self.a.expand(), self.b.expand(), self.c.expand())
+    }
+
+    /// Returns a new [`Triangle`],
+    /// with each coordinate value rounded towards the center of the [`Triangle`]
+    /// to the nearest integer, unless they are already an integer.
+    /// That is to say this function will return the biggest possible [`Triangle`]
+    /// with integer coordinates that is a subset of `self`.
+    #[inline]
+    pub fn trunc(self) -> Self {
+        Self::new(self.a.trunc(), self.b.trunc(), self.c.trunc())
+    }
+
+    /// Scales the [`Triangle`] by a `factor`
+    #[inline]
+    pub fn scale(self, factor: f64) -> Self {
+        Self::new(
+            (self.a.to_vec2() * factor).to_point(),
+            (self.b.to_vec2() * factor).to_point(),
+            (self.c.to_vec2() * factor).to_point(),
+        )
+    }
+
+    /// Returns the [`Ellipse`] that is bounded by this [`Triangle`].
+    #[inline]
+    pub fn to_ellipse(self) -> Ellipse {
+        Ellipse::from_triangle(self)
+    }
+
+    /// Is this [`Triangle`] finite?
+    #[inline]
+    pub fn is_finite(&self) -> bool {
+        self.a.is_finite() && self.b.is_finite() && self.c.is_finite()
+    }
+
+    /// Is this [`Triangle`] NaN?
+    #[inline]
+    pub fn is_nan(&self) -> bool {
+        self.a.is_nan() || self.b.is_nan() || self.c.is_nan()
+    }
+
+    /// [`Triangle`]'s vertices as an array of Points
+    #[inline]
+    pub fn as_array(&self) -> [Point; 3] {
+        [self.a, self.b, self.c]
+    }
+}
+
+impl From<(Point, Point, Point)> for Triangle {
```

Comment:
There is no corresponding conversion for `Line`, but maybe this is something that should be revisited.

Reply:
I was gonna do something like this for `Line`:

```Rust
impl From<(Line, Line)> for Triangle {
    fn from(lines: (Line, Line)) -> Triangle {
        Triangle::new(lines.0.p0, lines.0.p0, lines.1.p0)
    }
}
```

but this just ignores the annoying cases like if the two lines dont meet at a point

Reply:
This is a different question. I believe that Raph is referring to the api:
```rust
impl From<(Point, Point)> for Line {...}
```

Reply:
Oh right that makes a bit more sense, sorry. This is already implemented for rect which is why i did it here for triangle, I feel like it should therefore be implemented for `Line` to keep some continuity

Reply:
I've added a `From<(Point, Point)> for Line` in #376.